sklearn/tree/_tree.pyx

```diff
@@ -95,48 +96,129 @@ NODE_DTYPE = np.dtype({
 # =============================================================================
 
 cdef class Criterion:
-    """Interface for impurity criteria."""
+    """Interface for impurity criteria. 
+
+    This object stores methods on how to calculate how good a split is using 
+    different metrics.
+    """
 
     cdef void init(self, DOUBLE_t* y, SIZE_t y_stride, DOUBLE_t* sample_weight,
                    double weighted_n_samples, SIZE_t* samples, SIZE_t start,
                    SIZE_t end) nogil:
-        """Initialize the criterion at node samples[start:end] and
-           children samples[start:start] and samples[start:end]."""
+        """Placeholder for a method which will initialize the criterion.
+
+        Parameters
+        ----------
+        y: array-like, dtype=DOUBLE_t
+            This is a vector of targets in a memory-efficient buffer format, 
+            used to evaluate the split
+        y_stride: SIZE_t
+            This indicates the stride of the buffer to allow access to elements
+            easily
+        sample_weight: array-like, dtype=DOUBLE_t
+            The weight of each sample
+        weighted_n_samples: DOUBLE_t
+            The total weight of the samples being considered
+        samples: array-like, dtype=DOUBLE_t
+            The input data which is being split
+        start: SIZE_t
+            The first sample to be used on this leaf
+        end: SIZE_t
+            The last sample used in this leaf
```

Comment:
This makes it sounds like it is included, but it is not. Node samples are `samples[start:end]`.